### PR TITLE
Add DeepRVAT annotation pipeline

### PR DESCRIPTION
**Pull request by endast**
# What 
Added Support for all models needed for basic deeprvat run: AbSplice, CADD, SpliceAI, PrimateAI, DeepRIPE (hg2, k5 and parclip model) and DeepSEA. Performs PCA on DeepSEA annotations and merges the annotations into a single file. The pipeline uses BCF files created by the preprocessing steps as input, as well as a pvcf.txt file explaining the order of input files. The pipeline downloads and installs needed repositories, but some files have to be downloaded manually. See [VEP Plugins page](https://www.ensembl.org/info/docs/tools/vep/script/vep_plugins.html) for instructions to download data needed for CADD, SpliceAI and PrimateAI Plugins.

263267336-d483831e-3558-4e21-9845-4b62ad4eecc3

### CODE REVIEW

File: deeprvat/annotations/annotations.py [hunk 1]
```diff
@@ -0,0 +1,1299 @@
...
+        encoded_seqs = np.transpose(encoded_seqs, axes=(0, 2, 1))
+
+        return encoded_seqs
+
+
+def deepripe_score_variant_onlyseq_all(
+    model_group, variant_bed, genomefasta, seq_len=200, batch_size=1024, n_jobs=32
+):
+    predictions = {}
+    # counter = 0
```
Comment (endast) [inline deeprvat/annotations/annotations.py:448]: Can we remove this code here?

File: pipelines/config/deeprvat_annotation_config.yaml [hunk 2]
```diff
@@ -1,29 +1,37 @@
 fasta_dir : reference
-fasta_file_name : GRCh38.primary_assembly.genome.fa.gz
+fasta_file_name : hg38.fa
 bcftools_load_cmd : module load bcftools/1.9
```
Comment (endast) [inline pipelines/config/deeprvat_annotation_config.yaml]: Comment out this and add explanation

File: pipelines/config/deeprvat_annotation_config.yaml [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
 htslib_load_cmd : module load htslib/1.9
 perl_load_cmd : module load perl/5.20.2
 vep_load_cmd : module load vep/108.1
 
-vcf_file_pattern : ukb23156_c{chr}_b{block}_v1
+source_variant_file_pattern : test_vcf_data_c{chr}_b{block}
+source_variant_file_type: 'vcf.gz'
 included_chromosomes : ['21','22']
```
Comment (endast) [inline pipelines/config/deeprvat_annotation_config.yaml]: Comment out explain use

File: pipelines/preprocess.snakefile [hunk 4]
```diff
@@ -81,6 +81,7 @@ rule combine_genotypes:
             chr=chromosomes,
             block=block,
         ),
+    resources: mem_mb=15000
```
Comment (endast) [inline pipelines/preprocess.snakefile]: remove this

File: pipelines/preprocess.snakefile [hunk 5]
```diff
@@ -132,6 +133,7 @@ rule preprocess:
         qc_filtered_samples=qc_filtered_samples_dir,
     output:
         expand(preprocessed_dir / "genotypes_chr{chr}.h5",chr=set(chromosomes)),
+    resources: mem_mb=15000
```
Comment (endast) [inline pipelines/preprocess.snakefile]: remove this

File: pipelines/resources/absplice_config.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+# parameters for variant filtering
+variant_filtering:
+  minor_allele_frequency: 
+    apply_filter: False
+    filter_cutoff: 0.001
+  max_num_samples: 
+    apply_filter: False
+    filter_cutoff: 2
+
+# False by default, you can change this if you want to run on large datasets (see Readme)
```
Comment (endast) [inline pipelines/resources/absplice_config.yaml:18]: see absplice read me :)

**Review by Marcel-Mueck — APPROVED** [review]
Smoketest, and example run worked fine, everything seems to work.

**Pull request merged**